components/definition/cellRules/sections/mechanisms/NetPyNENewMechanism.js

```diff
@@ -14,56 +32,70 @@ export default class NetPyNENewMechanism extends React.Component {
       open: false,
       mechanisms: []
     };
-    this.handleClick = this.handleClick.bind(this);
-    this.handleButtonClick = this.handleButtonClick.bind(this);
   };
   
-  componentDidMount() {
-    Utils
-      .evalPythonMessage("netpyne_geppetto.getAvailableMechs", [])
-      .then((response) => {
-        var menuItems = []
-        response.forEach((item) => 
-          menuItems.push(<MenuItem id={item} key={item} value={item} primaryText={item}/>)
-        )
-        this.setState({mechanisms: menuItems})
-      })
+  async componentDidMount() {
+    const response = await Utils.evalPythonMessage("netpyne_geppetto.getAvailableMechs", [])
+    this.setState({mechanisms: response})
   };
   
-  handleButtonClick = (event) => {
-    // This prevents ghost click.
-    event.preventDefault();
-    this.setState({
-      open: true,
-      anchorEl: event.currentTarget,
-    });
+  handleClick = (value) => {
+    this.setState({open: false});
+    this.props.handleClick(value);
   };
 
-  handleRequestClose = () => {
-    this.setState({
-      open: false,
-    });
-  };
-  
-  handleClick(event, value) {
-    this.handleRequestClose();
-    this.props.handleClick(value);
+  handleButtonClick = (anchor) => {
+    const { blockButton, handleHierarchyClick } = this.props;
+    if (!blockButton) {
+      this.setState({open: true, anchorEl: anchor})
+    };
+    handleHierarchyClick();
   };
 
   render() {
+    const { disabled, blockButton } = this.props;
+    const { open, anchorEl, mechanisms } = this.state;
+    
     return <div>
-      <FloatingActionButton id={"addNewMechButton"} mini={true} style={{ margin: 10, float: 'left'}} onClick={this.handleButtonClick}>
-        <ContentAdd />
-      </FloatingActionButton>
+      <IconButton
+        data-tooltip={
+          disabled 
+            ? "No section selected" 
+            : blockButton 
+              ? "Explore mechanisms" 
+              : "Add new mechanism"
+        }
+        id="newMechButton"
+        className="gearAddButton"
+        disabled={disabled}
+        onClick={ e => this.handleButtonClick(e.currentTarget) }
+      >
+        <FontIcon 
+          style={{position: 'absolute'}}
+          className="gpt-fullgear"
+          color={changeColor} 
+          hoverColor={hoverColor} 
+        />
+        
+        {disabled 
```

Comment:
Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?